File: alikin-frontend/src/app/communities/community-response.ts

export interface UserBasicResponse {
  id: number;
  name: string;
  nickname: string;
  profilePictureUrl: string | null;
}


export interface CommunityRadio {
  name: string;
  streamUrl: string;
  logoUrl: string | null;
}


export interface CommunityResponse {
  id: number;
  name: string;
  description: string | null;
  imageUrl: string | null;
  createdAt: string;
  leader: UserBasicResponse;
  membersCount: number;
  userRole: 'LEADER' | 'MEMBER' | 'VISITOR' | null;


  radioPlaylist: CommunityRadio | null;





  radioStationName?: string;
  radioStreamUrl?: string;
  radioStationLogoUrl?: string;

  member?: boolean;

}

File: Alikin-backend/src/main/java/com/backendalikin/dto/response/CommunityResponse.java

package com.backendalikin.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CommunityResponse {
    private Long id;
    private String name;
    private String description;
    private String imageUrl;
    private LocalDateTime createdAt;
    private UserBasicResponse leader;
    private int membersCount;
    private boolean isMember; 
    private String userRole; 
    private PlaylistBasicResponse radioPlaylist;
    private String radioStationName;
    private String radioStreamUrl;
    private String radioStationLogoUrl;
}